dbms/src/Flash/Coprocessor/DAGExpressionAnalyzerHelper.cpp

```diff
@@ -123,6 +123,31 @@ String DAGExpressionAnalyzerHelper::buildIfNullFunction(
     return analyzer->applyFunction(func_name, argument_names, actions, getCollatorFromExpr(expr));
 }
 
+String DAGExpressionAnalyzerHelper::buildNullEqFunction(
+    DAGExpressionAnalyzer * analyzer,
+    const tipb::Expr & expr,
+    const ExpressionActionsPtr & actions)
+{
+    /// nullEq(col1, col2) == isNull(col1) AND isNull(col2) OR coalesce(Equal(col1, col2), 0)
```

Comment:
It seems that `isNull(col1) AND isNull(col2) OR coalesce(Equal(col1, col2), 0)` can be reduced to `Equal(col1, col2)` when col1 and col2 is not null.

Reply:
You can use `actions->getSampleBlock().getByName(col).type->isNullable()` to determine if col is nullable